src/lib.rs

```diff
@@ -92,6 +92,38 @@ impl StrRange {
         self.end
     }
 
+    /// A range with an adjusted end.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `end` is before this range's `start`.
+    pub fn with_start(self, start: StrIndex) -> StrRange {
+        StrRange::between(start, self.end())
+    }
+
+    /// A range with an adjusted end.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `end` is before this range's `start`.
+    pub fn with_end(self, end: StrIndex) -> StrRange {
+        StrRange::between(self.start(), end)
+    }
+
+    /// A range starting at this range with length `len`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use str_index::*;
+    /// let point = StrIndex::from(10);
+    /// let range = StrRange::from(point).with_len(10.into());
+    /// assert_eq!(range, StrRange::between(10.into(), 20.into()));
+    /// ```
+    pub fn with_len(self, len: StrIndex) -> StrRange {
```

Comment:
-0, as it’s not too obvious, from the name of the function what would be the start